src/toast/utils.py

```diff
@@ -36,6 +36,64 @@
 )
 
 
+def set_numba_threading():
+    """Set the numba threading layer.
+
+    For parallel numba jit blocks, the backend threading layer is selected at runtime
+    based on an order set inside the numba package.  We would like to change the
+    order of selection to prefer the OpenMP backend in order to be compatible with
+    other pieces of compiled code that use the OpenMP thread pool.  If OpenMP is not
+    supported, then we next try to use TBB for threading and finally fall back to the
+    default, which uses a multiprocessing workqueue.
+
+    Args:
+        None
+
+    Returns:
+        None
+
+    """
+    log = Logger.get()
+    threading = "default"
+    try:
+        from numba.npyufunc import omppool
+
+        threading = "omp"
+    except ImportError:
+        # no OpenMP support
+        try:
+            from numba.npyufunc import tbbpool
+
+            threading = "tbb"
+        except ImportError:
+            # no TBB
+            pass
+    try:
+        from numba import config, njit, threading_layer
+
+        config.THREADING_LAYER = threading
+
+        # In order to get numba to actually select a threading layer, we must
+        # trigger compilation of a parallel function.
+        @njit(parallel=True)
+        def foo(a, b):
+            return a + b
+
+        x = np.arange(10.0)
+        y = x.copy()
+        out_py = x + y
+        out_numba = foo(x, y)
+        if not np.allclose(out_py, out_numba):
```

Comment:
It's unusual to find a case where Numba successfully compiled something (via a specified signature or otherwise) and the resulting compiled function returns an incorrect result. If this is the case it's a Numba bug, please file a bug report!